snapshot_dbg_cli_tests/test_cli_services.py

```diff
@@ -0,0 +1,540 @@
+    self.gcloud_service_mock.config_get_project = MagicMock(
+        return_value='cli-test-project')
+    self.gcloud_service_mock.get_access_token = MagicMock(
+        return_value='cli-test-access-token')
+
+    expected_is_debug_enabled = False
+
+    cli_services = CliServices(args)
+
+    self.assertEqual(args, cli_services.args)
+    self.assertEqual('foo@bar.com', cli_services.account)
+    self.assertEqual('cli-test-project', cli_services.project_id)
+    self.assertEqual('cli-test-access-token', cli_services.access_token)
+    self.assertEqual(self.http_service_mock, cli_services.http_service)
+    self.assertEqual(self.permissions_service_mock,
+                     cli_services.permissions_service)
+    self.assertEqual(self.firebase_management_service_mock,
+                     cli_services.firebase_management_service)
+
+    self.data_formatter_class_mock.assert_called_once()
+    self.user_output_class_mock.assert_called_once_with(
+        expected_is_debug_enabled, self.data_formatter_mock)
+    self.gcloud_service_class_mock.assert_called_once_with(
+        self.user_output_mock)
+
+    self.http_service_class_mock.assert_called_once_with(
+        project_id='cli-test-project',
+        access_token='cli-test-access-token',
+        user_output=self.user_output_mock)
+
+    self.permissions_service_class_mock.assert_called_once_with(
+        project_id='cli-test-project',
+        http_service=self.http_service_mock,
+        access_token='cli-test-access-token',
+        user_output=self.user_output_mock)
+
+    self.firebase_management_service_class_mock.assert_called_once_with(
+        http_service=self.http_service_mock,
+        project_id='cli-test-project',
+        user_output=self.user_output_mock)
+
+    self.firebase_rtdb_service_class_mock.assert_not_called()
+    self.snapshot_debugger_rtdb_service_class_mock.assert_not_called()
+
+  def test_constructor_debug_works_as_expected(self):
+    args_debug_not_set = argparse.Namespace()
+
+    args_debug_set_false = argparse.Namespace()
+    args_debug_set_false.debug = False
+
+    args_debug_set_true = argparse.Namespace()
+    args_debug_set_true.debug = True
+
+    testcases = [
+        ('Not set', args_debug_not_set, False),
+        ('Set false', args_debug_set_false, False),
+        ('Set true', args_debug_set_true, True),
+    ]
+
+    for test_name, args, expected_is_debug_enabled in testcases:
+      with self.subTest(test_name):
+        self.user_output_class_mock.reset_mock()
+        CliServices(args)
+
+        self.user_output_class_mock.assert_called_once_with(
+            expected_is_debug_enabled, self.data_formatter_mock)
+
+  def test_get_snapshot_debugger_rtdb_service_url_provided(self):
+    args = argparse.Namespace()
+
+    service = CliServices(args).get_snapshot_debugger_rtdb_service(
+        database_url='https://fake-url.firebaseio.com')
+
+    self.assertEqual(service, self.snapshot_debugger_rtdb_service_mock)
```

Comment:
Is this making sure that the mocking worked or is there something more nuanced at play?